dcpy/lifecycle/package/generate_metadata_assets.py

```diff
@@ -12,12 +14,38 @@
 DEFAULT_DATA_DICTIONARY_STYLESHEET_PATH = RESOURCES_PATH / "data_dictionary.css"
 
 
+def _render_html_template(
+    template_path: Path, template_vars: dict, stylesheet_path: Path | None = None
+) -> str:
+    with open(template_path, "r") as f:
+        template_text = f.read()
+    html = Template(template_text).render(template_vars)
+
+    if stylesheet_path:
+        with open(stylesheet_path, "r") as f:
+            css = f.read()
+    else:
+        css = ""
+    styled_html = css_inline.inline_fragment(html, css)
+
+    return styled_html
+
+
+def _compose_html_document(template_path: Path, fragments: list[str]) -> str:
+    with open(template_path, "r") as f:
+        template_text = f.read()
+    html = Template(template_text).render({"fragments": fragments})
```

Comment:
this concept of `fragments` maybe isn't as obvious as it should be, so maybe the function needs a better name?

Reply:
I think fragments is fine